gulpfile.js

```diff
@@ -71,422 +83,496 @@ const buildChanges = {
+/**
+ * Run test once and exit
+ */
+gulp.task('test', async (done) => {
+  const testPath = `${__dirname}${g3w.test.path}`;
+  const testGroupFolders = fs.readdirSync(testPath).filter(file => file !== 'group_template' && fs.statSync(testPath + '/' +file).isDirectory());
+  for (let i = 0; i < testGroupFolders.length; i++) {
+    await new Promise((resolve) => {
+      new karma.Server({
+        configFile: `${testPath}${testGroupFolders[i]}/karma.config.js`,
+        singleRun: true
+      },() => { resolve() }).start();
+    });
+  }
+  done();
 });
 
-gulp.task('g3w-admin-plugins',function() {
-  gulp.src(`${distFolder}/**/js/plugin.js`)
-    .pipe(rename(function(path){
-      const dirname = path.dirname;
-      const pluginname = dirname.replace('/js','');
-      path.dirname = `${conf.g3w_admin_paths[g3w_admin_version].g3w_admin_plugins_basepath}/${pluginname}/static/${pluginname}/js/`;
+/**
+ * Deafult development task (BrowserSync server)
+ */
+gulp.task('dev', ['build_external_assets', 'clean:dist', 'browserify:app', 'assets', 'watch', 'plugins', 'browser-sync']);
+
+/**
+ * Expose version of "package.json" without including whole file in published bundle,
+ * this happens because each ESM `import` is actually transformed into a CJS `require()`
+ * (NB: native ESM modules will not suffer of this security issue due to tree shaking)
+ *
+ * @see https://github.com/g3w-suite/g3w-client/issues/
+ * @see src\app\constant::APP_VERSION
+ * @see src\app\version
+ */
+gulp.task('version', function () {
+  fs.writeFileSync('src/version.js', `/* WARNING: this file is autogenerated by gulpfile.js, please do not edit manually */\n\nexport default '${packageJSON.version}';`);
+});
+
+// Backward compatibilities (v3.x)
+gulp.task('clean',                               ['clean:dist']);
+gulp.task('clean_vendor_node_modules_min',       ['clean:vendor']);
+gulp.task('cleanup',                             ['clean:app']);
+gulp.task('g3w-admin-client:clear',              ['clean:admin']);
+gulp.task('sethasvalues',                        ['md5hash']);
+gulp.task('vendor_node_modules_js',              ['__browserify_vendor_js']);
+gulp.task('concatenate_node_modules_vendor_min', ['browserify:vendor']);
+gulp.task('browserify',                          ['browserify:app']);
+gulp.task('add_external_resources_to_main_html', ['build_external_assets']);
+gulp.task('html',                                ['html:dev']);
+gulp.task('html:compiletemplate',                ['html:prod']);
+gulp.task('g3w-admin-plugins-select',            ['g3w-admin:plugins']);
+gulp.task('g3w-admin-client:static',             ['g3w-admin:static']);
+gulp.task('g3w-admin-client:template',           ['g3w-admin:templates']);
+
+//////////////////////////////////////////////////////////////////////////
+/**
+ * TODO: check which of the following tasks is actually required
+ */
+//////////////////////////////////////////////////////////////////////////
+
```

Comment:
@volterra79 can you check this before merge?

Reply:
ok .. i let you know as soon as possible